### ISSUE TITLE
test_xapian fails

### ISSUE THREAD

**Issue by mitya57** [sphinx-doc/sphinx#3246]
With the latest stable branch, I get this error when running the testsuite:

```
======================================================================
ERROR: test_searchadapters.test_xapian
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/dmitry/upstream/sphinx/sphinx/builders/html.py", line 724, in index_page
    self.indexer.feed(pagename, filename, title, doctree)
  File "/home/dmitry/upstream/sphinx/sphinx/websupport/search/__init__.py", line 48, in feed
    self.add_document(pagename, filename, title, doctree.astext())
TypeError: add_document() takes 4 positional arguments but 5 were given

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/nose/case.py", line 198, in runTest
    self.test(*self.arg)
  File "/home/dmitry/upstream/sphinx/tests/util.py", line 138, in skipper
    return test(*args, **kwds)
  File "/home/dmitry/upstream/sphinx/tests/util.py", line 138, in skipper
    return test(*args, **kwds)
  File "/home/dmitry/upstream/sphinx/tests/test_searchadapters.py", line 63, in test_xapian
    search_adapter_helper('xapian')
  File "/home/dmitry/upstream/sphinx/tests/test_searchadapters.py", line 31, in search_adapter_helper
    support.build()
  File "/home/dmitry/upstream/sphinx/sphinx/websupport/__init__.py", line 126, in build
    app.build()
  File "/home/dmitry/upstream/sphinx/sphinx/application.py", line 333, in build
    self.builder.build_update()
  File "/home/dmitry/upstream/sphinx/sphinx/builders/__init__.py", line 251, in build_update
    'out of date' % len(to_build))
  File "/home/dmitry/upstream/sphinx/sphinx/builders/__init__.py", line 322, in build
    self.write(docnames, list(updated_docnames), method)
  File "/home/dmitry/upstream/sphinx/sphinx/builders/__init__.py", line 360, in write
    self._write_serial(sorted(docnames), warnings)
  File "/home/dmitry/upstream/sphinx/sphinx/builders/__init__.py", line 367, in _write_serial
    self.write_doc_serialized(docname, doctree)
  File "/home/dmitry/upstream/sphinx/sphinx/builders/websupport.py", line 79, in write_doc_serialized
    self.index_page(docname, doctree, title)
  File "/home/dmitry/upstream/sphinx/sphinx/builders/html.py", line 727, in index_page
    self.indexer.feed(pagename, title, doctree)
TypeError: feed() missing 1 required positional argument: 'doctree'
```

This is with Python 3.5.2+ and xapian-bindings 1.4.1 on Debian unstable.

**Comment by tk0miya** [comment]
Thank you for reporting.
OMG, the xapian searcher was not modified at d27386cc95772a7e7cc47941a0c08b27e0272f67...

**Comment by tk0miya** [comment]
Fixed at 	cf79589.
Thank you for reporting.
